src/components/illustration/illustration.tsx

```diff
@@ -0,0 +1,57 @@
+import IllustrationSet, { IllustrationType } from './illustration-set';
+import React from 'react';
+import styled, { css } from 'styled-components';
+
+export enum IllustrationSize {
+  NORMAL = 'normal', // 72px
+  LARGE = 'large', // 144px
+  INHERIT = 'inherit', // 100% of the container
+}
+
+export interface IllustrationProps {
+  /**
+   * Pick an illustration by name.
+   */
+  show: IllustrationType;
+  /**
+   * Choose the size. Defaults to 72x72px, can be large (172x172px) or grow with the container.
+   */
+  size?: IllustrationSize;
+}
+
+const IllustrationLayout = styled.div<Pick<IllustrationProps, 'size'>>`
```

Comment:
This usage of `Pick<>` is impressive... but seems like overkill here to be honest. What's the benefit of keeping this?

Reply:
I only use the property `show` in this definition. Would you expect me to use an isolated type with a single property name or just pass in the `IllustrationProps`?

Reply:
By the way, I didn't want to randomly use an impressive typescript tool. `Pick` and the counterpart `Omit` are some great utility types to know and use (wisely) in your daily work 💪

Reference: https://www.typescriptlang.org/docs/handbook/utility-types.html

Reply:
I would have expected you to just pass in `IllustrationProps`. I don't see the benefit of using `Pick` here. The props are getting destructured and you are only pulling out the `size` property anyways.
Another point: This styled component will receive all the properties of `IllustrationProps` so this generic type might even be considered wrong since it hides away other properties of `IllustrationProps` that are very well available in the scope of the styled component.
My main concern is that it will confuse more juniory developers and make them wonder why this is used here and nowhere else.

This is clearly not a blocker and I'm fine with keeping it so I'll leave it to you to decide.

Reply:
Regarding  this quote:

> might even be considered wrong since it hides away other properties of IllustrationProps that are very well available in the scope of the styled component.

I think we have a misunderstanding. I try to clarify this but thanks for unblocking this issue 👍

---

Here a quick snippet from the related code (simplified)

```
export const Illustration = (props: IllustrationProps) => {
  return (
    <IllustrationLayout size={props.size}> // <-- our styled component with the props in discussion
      <IllustrationSvg />
    </IllustrationLayout>
  );
};
```

You can see that the component `IllustrationLayout` has the following props definition:

```
interface IllustrationLayout {
   size: IllustrationSize;
}
```

which is different (a actual subset) from the props coming in from the outside, being the full `IllustrationProps`:

```
export interface IllustrationProps {
  show: IllustrationType;
  size?: IllustrationSize;
}
```

That's why this pick make sense. If I would use `IllustrationProps` on the styled component definition (`const IllustrationLayout = styled.div<IllustrationProps>`) I would suggest that there are other properties like `size` which is wrong.

Reply:
Got it. You are right. I was assuming (without double checking, my bad) that you just did `{...props}` on this component.

Reply:
I did before for the styled components experiment ☝️ Riddle solved 😁

See my explanation:
https://github.com/satellytes/satellytes.com/pull/190#discussion_r750896607